src/hypothesis/core.py

```diff
@@ -501,13 +427,108 @@ def timed_test(*args, **kwargs):
-        if self.__in_final_replay:
-            return base
+    def execute(
+        self, data,
+        print_example=False,
+        is_final=False,
+        expected_failure=None,
+    ):
+        text_repr = [None]
+        if self.settings.deadline is None:
+            test = self.test
         else:
-            return base * 1.25
+            @proxies(self.test)
+            def test(*args, **kwargs):
+                self.__test_runtime = None
+                initial_draws = len(data.draw_times)
+                start = benchmark_time()
+                result = self.test(*args, **kwargs)
+                finish = benchmark_time()
+                internal_draw_time = sum(data.draw_times[initial_draws:])
+                runtime = (finish - start - internal_draw_time) * 1000
+                self.__test_runtime = runtime
+                if self.settings.deadline is not_set:
+                    if (
+                        not self.__warned_deadline and
+                        runtime >= 200
+                    ):
+                        self.__warned_deadline = True
+                        note_deprecation((
+                            'Test took %.2fms to run. In future the default '
+                            'deadline setting will be 200ms, which will '
+                            'make this an error. You can set deadline to '
+                            'an explicit value of e.g. %d to turn tests '
+                            'slower than this into an error, or you can set '
+                            'it to None to disable this check entirely.') % (
+                                runtime, ceil(runtime / 100) * 100,
+                        ))
+                else:
+                    current_deadline = self.settings.deadline
+                    if not is_final:
+                        current_deadline *= 1.25
+                    if runtime >= current_deadline:
+                        raise DeadlineExceeded(runtime, self.settings.deadline)
+                return result
+
+        def run(data):
+            with self.settings:
+                with BuildContext(data, is_final=is_final):
+                    import random as rnd_module
+                    rnd_module.seed(0)
+                    args, kwargs = data.draw(self.search_strategy)
+                    if expected_failure is not None:
+                        text_repr[0] = arg_string(test, args, kwargs)
+
+                    if print_example:
+                        report(
+                            lambda: 'Falsifying example: %s(%s)' % (
+                                test.__name__, arg_string(test, args, kwargs)))
+                    elif current_verbosity() >= Verbosity.verbose:
+                        report(
+                            lambda: 'Trying example: %s(%s)' % (
+                                test.__name__, arg_string(test, args, kwargs)))
+
+                    collector = self.collector
```

Comment:
Cleaner to just refer to `self.collector` in the next few lines without the assignment.